airsonic-main/src/main/java/org/airsonic/player/service/search/AnalyzerFactory.java

```diff
@@ -47,75 +55,128 @@
+        return builder;
+    }
+
+    private Builder createKeywordAnalyzerBuilder() throws IOException {
+        return CustomAnalyzer.builder()
+                .withTokenizer(KeywordTokenizerFactory.class);
+    }
+
+    private Builder createGenreAnalyzerBuilder() throws IOException {
+        Builder builder = createKeywordAnalyzerBuilder();
+        addTokenFilterForTokenToDomainValue(builder);
+        return builder;
+    }
 
     /**
-     * Return analyzer.
+     * Returns the Analyzer to use when generating the index.
+     * 
+     * Whether this analyzer is applied to input values ​​depends on
+     * the definition of the document's fields.
      * 
      * @return analyzer for index
+     * @see DocumentFactory
      */
     public Analyzer getAnalyzer() {
-        if (null == analyzer) {
-            analyzer = new CustomAnalyzer();
+        if (isEmpty(analyzer)) {
+            try {
+                analyzer = createDefaultAnalyzerBuilder().build();
+            } catch (IOException e) {
+                LOG.error("Error when initializing Analyzer.", e);
+            }
         }
         return analyzer;
     }
 
     /**
-     * Return analyzer.
+     * Returns the analyzer to use when generating a query for index search.
      * 
-     * @return analyzer for index
+     * String processing handled by QueryFactory
+     * is limited to Lucene's modifier.
+     * 
+     * The processing of the operands is expressed
+     * in the AnalyzerFactory implementation.
+     * Rules for tokenizing/converting input values ​
+     * should not be described in QueryFactory.
+     * 
+     * @return analyzer for query
+     * @see QueryFactory
      */
     public Analyzer getQueryAnalyzer() {
-        if (null == queryAnalyzer) {
-            queryAnalyzer = new CustomAnalyzer();
-        }
-        return queryAnalyzer;
-    }
+        if (isEmpty(queryAanalyzer)) {
+            try {
 
-    /*
-     * The legacy CustomAnalyzer implementation is kept as it is.
-     */
-    private class CustomAnalyzer extends StandardAnalyzer {
-        private CustomAnalyzer() {
-            /*
-             * Version.LUCENE_30
-             * It is a transient description and will be deleted when upgrading the version.
-             * SearchService variables are not used because the reference direction conflicts.
-             */
-            super(Version.LUCENE_30);
-        }
+                Analyzer defaultAnalyzer = createDefaultAnalyzerBuilder().build();
+                Analyzer genreAnalyzer = createGenreAnalyzerBuilder().build();
 
-        @Override
-        public TokenStream tokenStream(String fieldName, Reader reader) {
-            TokenStream result = super.tokenStream(fieldName, reader);
-            return new ASCIIFoldingFilter(result);
-        }
+                Map<String, Analyzer> fieldAnalyzers = new HashMap<>();
+                fieldAnalyzers.put(FieldNames.GENRE, genreAnalyzer);

```

Comment:
Commenting out this process has no effect on existing web screens or clients.
Therefore, it may be considered unnecessary processing, but it is a relatively important decision in the specification.

Reply:
The reason for stricting the genre string is to change the domain value from the previous mutable to immutable.
This means that genre reference on DB is possible via domain value.

We can cope with difficult requirements like #105.
While multivalue processing on DB involves very complex implementations, doing this with Lucene is relatively safe and fast.

Conversely, the requirement to change domain design from immutable to mutable, should be cautious.
As in #1132.